solvebio/resource/object.py

```diff
@@ -0,0 +1,69 @@
+"""Solvebio Object API resource"""
+from ..help import open_help
+from ..client import client
+
+from .solveobject import convert_to_solve_object
+from .apiresource import CreateableAPIResource
+from .apiresource import ListableAPIResource
+from .apiresource import SearchableAPIResource
+from .apiresource import UpdateableAPIResource
+from .apiresource import DeletableAPIResource
+
+
+class Object(CreateableAPIResource,
+             ListableAPIResource,
+             DeletableAPIResource,
+             SearchableAPIResource,
+             UpdateableAPIResource):
+    """
+    An object is a resource in a Vault.  It has three possible types,
+    though more may be added later: folder, file, and SolveBio Dataset.
+    """
+    RESOURCE_VERSION = 2
+
+    LIST_FIELDS = (
+        ('id', 'ID'),
+        ('object_type', 'Object Type'),
+        ('path', 'Path'),
+        ('filename', 'Filename'),
+        ('description', 'Description'),
+    )
+
+    @classmethod
+    def retrieve_by_full_path(cls, full_path, **params):
```

Comment:
This is probably a bad idea, but maybe retrieve() can take full_path=X as a kwarg.